lib/wraft_doc/search_engine/typesense_index.ex

```diff
@@ -0,0 +1,84 @@
+defprotocol WraftDoc.SearchEngine.TypesenseIndex do
+  @moduledoc """
+  Protocol defining the interface for converting records to Typesense documents.
+  """
+
+  @doc """
+  Converts a struct to a map format suitable for Typesense indexing.
+  """
+  @spec to_document(t()) :: map()
+  def to_document(struct)
+
+  @doc """
+  Returns the collection name for the given struct type.
+  """
+  @spec collection_name(t()) :: String.t()
+  def collection_name(struct)
+
+  @doc """
+  Returns the schema configuration for the collection.
+  """
+  @spec collection_schema(t()) :: map()
+  def collection_schema(struct)
+end
+
+defimpl WraftDoc.SearchEngine.TypesenseIndex, for: WraftDoc.Document.ContentType do
+  @moduledoc """
+  Implementation of the Index protocol for ContentType schema.
+  Handles conversion of ContentType records for Typesense indexing.
+  """
+
+  alias WraftDoc.Document.ContentType
+
+  @doc """
+  Converts a ContentType struct to a map format suitable for Typesense indexing.
+  """
+  def to_document(%ContentType{} = content_type) do
+    %{
+      id: to_string(content_type.id),
+      record_type: "content_type",
+      name: content_type.name,
+      description: content_type.description || "",
```

Comment:
No need of the default value

```suggestion
      description: content_type.description,
```